### PR TITLE
refactor(chat): separate logic into hooks

### PR DESCRIPTION
**Pull request by iMADi-ARCH**
# Description

This PR separates requests and speech logic into hooks following the style of @mamadoudicko 😄

No extra features have been added.

## Checklist before requesting a review

Please delete options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented hard-to-understand areas
- [x] Any dependent changes have been merged

### CODE REVIEW

**Comment by iMADi-ARCH** [comment]
Dont merge before a review from @mamadoudicko

File: frontend/app/chat/hooks/useSpeech.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { isSpeechRecognitionSupported } from "../helpers/isSpeechRecognitionSupported";
+import { useQuestion } from "./useQuestion";
+
+export const useSpeech = () => {
+  const [isListening, setIsListening] = useState(false);
```
Comment (mamadoudicko) [inline frontend/app/chat/hooks/useSpeech.ts:6]: 😍

File: frontend/app/chat/hooks/useQuestion.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+
+import { useSupabase } from "@/app/supabase-provider";
+import { useBrainConfig } from "@/lib/context/BrainConfigProvider/hooks/useBrainConfig";
+import { useAxios } from "@/lib/useAxios";
+import { redirect } from "next/navigation";
+export const useQuestion = () => {
+  const [question, setQuestion] = useState("");
+  const [history, setHistory] = useState<Array<[string, string]>>([]);
```
Comment (mamadoudicko) [inline frontend/app/chat/hooks/useQuestion.ts:9]: 😍

File: frontend/app/chat/components/ChatMessages.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        </div>
+      ) : (
+        <AnimatePresence initial={false}>
+          {history.map(([speaker, text], idx) => {
+            return (
+              <ChatMessage
+                ref={idx === history.length - 1 ? lastChatRef : null}
+                key={idx}
+                speaker={speaker}
+                text={text}
```
Comment (mamadoudicko) [inline frontend/app/chat/components/ChatMessages.tsx:31]: To seeing if we can avoid using `index` as `key` here

**Review by mamadoudicko — CHANGES_REQUESTED** [review]
Great PR @iMADi-ARCH!

Really love the improved DX with `useSpeech` and `useQuestion` hooks. 

Kudos!

**Pull request merged**